diff -Naur scgi-1.2.orig/apache2/mod_scgi.c scgi-1.2/apache2/mod_scgi.c
--- scgi-1.2.orig/apache2/mod_scgi.c	2005-05-28 16:33:03.000000000 +0200
+++ scgi-1.2/apache2/mod_scgi.c	2005-05-29 00:00:42.000000000 +0200
@@ -131,7 +131,7 @@
     apr_size_t n;
     while (len > 0) {
         n = len;
-        if ((rv = apr_send(sock, buf, &n))) return rv;
+        if ((rv = apr_socket_send(sock, buf, &n))) return rv;
         buf += n;
         len -= n;
     }
@@ -191,8 +191,8 @@
     char *buf;
     int i;
     apr_status_t rv = 0;
-    apr_port_t  port = 0;
-    apr_sockaddr_port_get(&port, r->connection->remote_addr);
+    apr_port_t  port = r->connection->remote_addr->port;
+    //apr_sockaddr_port_get(&port, r->connection->remote_addr);
 
     log_debug(APLOG_MARK,r, "sending headers");
     /* CONTENT_LENGTH must come first and always be present */
@@ -308,7 +308,7 @@
     }
 
     *sock = NULL;
-    rv = apr_socket_create(sock, AF_INET, SOCK_STREAM, r->pool);
+    rv = apr_socket_create(sock, AF_INET, SOCK_STREAM, APR_PROTO_TCP, r->pool);
     if (rv) {
         log_err(APLOG_MARK, r, rv, "apr_socket_create() error");
         return rv;
